src/Lean/Widget/InteractiveGoal.lean

```diff
@@ -98,16 +126,22 @@ def addInteractiveHypothesisBundle (hyps : Array InteractiveHypothesisBundle) (i
     isType     := (← instantiateMVars type).isSort
   }
 
+open Meta in
+variable [MonadControlT MetaM n] [Monad n] [MonadError n] [MonadOptions n] [MonadMCtx n] in
+def withGoalCtx (goal : MVarId) (action : LocalContext → MetavarDecl → n α) : n α := do
```

Comment:
Should this be here or somewhere in `Meta`?

Reply:
I wasn't sure, it's an extraction of functionality of the method below, and it appeared to me to be doing an InteractiveGoal-specific thing of `sanitizeNames` which I didn't really understand so I kept it here.